homeassistant/components/powerwall/sensor.py

```diff
@@ -102,18 +104,21 @@ def device_class(self):
     @property
     def unique_id(self):
         """Device Uniqueid."""
-        return f"{self.base_unique_id}_{self._meter}_instant_power"
+        return f"{self.base_unique_id}_{self._meter.value}_instant_power"
 
     @property
     def state(self):
         """Get the current value in kW."""
-        meter = self._coordinator.data[POWERWALL_API_METERS][self._meter]
-        return round(float(meter.instant_power / 1000), 3)
+        return (
+            self._coordinator.data[POWERWALL_API_METERS]
+            .get(self._meter)
+            .get_power(precision=3)
```

Comment:
Need to verify this doesn't do I/O

Reply:
The `get` just accesses the meter from `MetersAggregatesResponse` and `get_power` returns the `instant_power` attribute of a `MetersResponse` convert to kw. So no I/O is performed.